variable/include/scipp/variable/variable.tcc

```diff
@@ -271,411 +51,131 @@ struct default_init<Eigen::Matrix<T, Rows, Cols>> {
 };
 } // namespace
 
-/// Implementation of VariableConcept that holds data.
-template <class T>
-class DataModel : public VariableConceptT<typename T::value_type> {
+/// Implementation of VariableConcept that holds and array with element type T.
+template <class T> class DataModel : public VariableConcept {
 public:
-  using value_type = std::remove_const_t<typename T::value_type>;
-
-  DataModel(const Dimensions &dimensions, T model,
-            std::optional<T> variances = std::nullopt)
-      : VariableConceptT<typename T::value_type>(dimensions),
-        m_values(
-            model ? std::move(model)
-                  : T(dimensions.volume(), default_init<value_type>::value())),
+  using value_type = T;
+
+  DataModel(const Dimensions &dimensions, element_array<T> model,
```

Comment:
I find this a lot clearer than what we had before.

Reply:
It is, isn't it. I think it only became possible thanks for a couple other smaller chnges and improvements, e.g., to transform we did over the past year.